File: 0792-binary-search/0792-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left=0
        right = len(nums) - 1
        
        while left <= right:
            mid = (left + right)// 2
            if nums[mid] == target:
                return mid  # Target found, return its index
            elif nums[mid] < target:
                left = mid + 1  # Move to the right half
            elif nums[mid]>target:
                right = mid - 1 # Move to the left half

        #  mid=(left+right)//2
        
        return -1  # Target not found

File: 0792-binary-search/0792-binary-search.cpp
class Solution {
public:
    int search(vector<int>& arr, int target) {
        int l=0;
        int h=arr.size()-1;
        int mid=(l+h)/2;

        while(l<=h){
            if(arr[mid]==target){
                return mid;
            }
            else if(arr[mid]<target){
                l=mid+1;
            }
            if(arr[mid]>target){
                h=mid-1;
            }
            mid=(l+h)/2;

        }
       return -1; 
    }
};



//--------------------using linear search ------------------------

/*



class Solution {
public:
    int search(vector<int>& nums, int target) {
        for(int i=0;i<nums.size();i++){
            if(nums[i]==target){
                return i;
            }
        }
        return -1;
    }
};
*/